graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,331 @@
+        # sorted so it returns the vertex with the highest probability first.
+        nomination_list = list(zip(b_inds[self.n_seeds :], P_outp[0]))
+        nomination_list.sort(key=lambda x: x[1], reverse=True)
+        self.nomination_list = np.array(nomination_list)
+
+        return self
+
+    def fit_predict(self, A, B, voi, seeds):
+        """
+        Fits model to two adjacency matrices and returns nomination list
+
+        Parameters
+        ----------
+        A: 2d-array, square
+            Adjacency matrix of `A`, the graph where voi is known
+
+        B: 2d-array, square
+            Adjacency matrix of `B`, the graph where voi is not known
+
+        voi: int
+            Vertex of interest (voi)
+
+        seeds: list
+            List of length two `[seedsA, seedsB]` where first element is
+            the seeds associated with adjacency matrix A
+            and the second element the adjacency matrix associated with B, note
+            `len(seedsA)==len(seedsB)` The elements of `seedsA` and `seedsB` are
+            vertices which are known to be matched, that is, `seedsA[i]` is matched
+            to vertex `seedsB[i]`.
+
+        Returns
+        -------
+        nomination_list : 2d-array
+            The nomination array
+        """
+        retval = self.fit(A, B, voi, seeds)
+
+        # It is possible that the algorithm stops early and returns none
+        # if the task is impossible. Note, this does not mean that the input
+        # arguments were erroneous.
+        if retval is None:
+            return None
+
+        return self.nomination_list
+
+
+def _get_induced_subgraph(graph_adj_matrix, order, node, mindist=1):
+    """
+    Generates a vertex list for the induced subgraph about a node with
+    max (order_ and min distance parameters.
+
+    Parameters
+    ----------
+    graph_adj_matrix: 2-d array
+        Adjacency matrix of interest.
+
+    order: int
+        Distance to create the induced subgraph with. Max distance away from
+        the node to include in subgraph.
+
+    node: int
+        The vertex to center the induced subgraph about.
+
+    mindist: int (default = 1)
+        The minimum distance away from the node to include in the subgraph.
+
+    Returns
+    -------
+    induced_subgraph : list
+        The list containing all the vertices in the induced subgraph.
+    """
+    # Note all nodes are zero based in this implementation, i.e the first node is 0
+    dists = [[node]]
+    dists_conglom = [node]
+    for ii in range(1, order + 1):
+        clst = []
+        for nn in dists[-1]:
+            clst.extend(list(np.where(graph_adj_matrix[nn] >= 1)[0]))
+        clst = np.array(list(set(clst)))
+
+        cn_proc = np.setdiff1d(clst, dists_conglom)
+
+        dists.append(cn_proc)
+
+        dists_conglom.extend(cn_proc)
+        dists_conglom = list(set(dists_conglom))
+
+    ress = itertools.chain(*dists[mindist : order + 1])
+
```

Comment:
i would suggest using something like `np.unique` in place of all of the `list(set())` stuff where it's appropriate